blocks/components/editable/index.js

```diff
@@ -101,14 +110,26 @@ export default class Editable extends wp.element.Component {
 		this.props.onFocus();
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	isActive() {
+		return tinymce.EditorManager.activeEditor.id === this.editor.id;
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		this.content = this.getContent();
+		this.selection = this.getSelection();
+
+		if ( ! this.props.onChange ) {
+			return;
+		}
+
+		if ( ! isEqual( this.props.value, this.content ) ) {
```

Comment:
>Hm, I was hoping this is something we'd handle in the reducers, or dispatch ADD_UNDO_LEVEL.

I might not be communicating well, but I agree with you on this. I just don't think we need each keystroke to trigger an update to state; we can choose different intervals instead (focus out, after a 5s debounce delay, every new paragraph added, etc). Not something we necessarily need to flesh out here or now.